conbench/numstr.py

```diff
@@ -52,9 +52,55 @@
 # import sigfig
 
 
+def numstr_dyn(v: Union[float, int]) -> str:
+    """
+    Turn number into string. Limit the count of significant figures.
+    Dynamically switch to scientific notation for large numbers (larger 10**6)
+    and small numbers (smaller 10**-6).
+
+    Regarding precision: this is so far meant to be used in
+    overview/summary/tabular representation where the 'rough' measurement
+    result should be presented in a canonical/non-cryptical way. Higher
+    precision / raw data is available to the user elsewhere. precision/sigfigs
+    is no argument to this function at this point to hopefully achieve
+    consistency across UI.
+
+    >>> numstr.numstr_dyn(1.290823987290392)
+    '1.29082'
+    >>> numstr.numstr_dyn(1239127)
+    '1.24e+06'
```

Comment:
note: since generating examples I have changed `exp_digits=1` (see below), i.e. this here would show `e+6` instead of `+06`. I did this to save horizontal space. Without losing horizontal alignment: the numbers we deal with should rarely exceed  9*10^9 and rarely be smaller than 10^-9.